app/policies/supported_permission_policy.rb

```diff
@@ -0,0 +1,16 @@
+class SupportedPermissionPolicy < BasePolicy
+  class Scope < ::BasePolicy::Scope
+    def resolve
+      if current_user.superadmin?
+        scope.all
+      elsif current_user.admin?
+        scope.all
+      elsif current_user.organisation_admin?
+        app_ids = Pundit.policy_scope(current_user, :manage_supported_permissions_for_application).pluck(:id)
```

Comment:
One returns a list of `Doorkeeper::Application`s (`ManageSupportedPermissionsForApplicationPolicy`), the other returns a list of `SupportedPermission` (`SupportedPermissionPolicy`).  The name for the former is far from ideal, so I'm open to suggestions.  I don't think it can just be `ApplicationPolicy` because it's not the list of applications the user can see, but the list of applications the user can see in a given context (arguably maybe `SupportedPermissionPolicy` should be given a more explicit name too.